.github/workflows/reproducible_build.yml

```diff
@@ -0,0 +1,33 @@
+name: Reproducible-build
+
+on:
+  push:
+    branches: [ main, 'release-*' ]
+  pull_request:
+    branches: [ main, 'release-*' ]
+
+  workflow_dispatch:
```

Comment:
Generally I think this type of PR checker would live as `job` inside of the `main.yml`  like: https://github.com/risc0/risc0/blob/main/.github/workflows/main.yml#L195 

Any reason here to have it broken off?